### DISCUSSION TITLE
the tutorial of project 0

### DISCUSSION THREAD

**Discussion by wtby111** [sist-cs120/project-wiki#32] [Project 1 Q&A] [answered]
After  configure directory to ASIO SDK of AudioLatencyDemo,

![CXRSN%`8FCL4MTK{ 1 5$)D](https://github.com/user-attachments/assets/dd6cb589-305e-4ecf-b54f-26f5a491c2d2)

following application run and it works

![~FHVFBDO3T2I5N@AIPF3IGI](https://github.com/user-attachments/assets/253bb93e-5afe-4d06-8fc4-737e78ac498e)

but there is no icon of ASIO driver
and in 02_write_a_player
 There is an error: member function declared with 'override' does not override a base class member		

![_GG8C2VZ)%ZLXOBT4$DAX20](https://github.com/user-attachments/assets/552c4092-2f9c-49e1-931d-62333d8333f4)

I tried to delete ‘override’ and ran it successfully but no sound.

the version of juce I'm using now is is v7.0.12. And I tried to use v8.0.1, but have the same problem

**Comment by SIST1C407** [comment, marked as answer]
It seems like that JUCE no longer supports the construction of an AudioIODeviceCallback object. However, our test with JUCE v7.0.0 has shown that the demo runs successfully.

**Reply by wtby111** [reply]
v7.0.0 seems ok but there was a new problem
![T$P}S 0M (1UAD%1_Q_H45K](https://github.com/user-attachments/assets/b9e1385d-d758-40ce-a4bb-09c4611947ac)
every input doesn't work

**Reply by SIST1C407** [reply]
Is your laptop produced by Lenovo? Based on the feedback we received last year, the ASIO driver may not be compatible with Lenovo's audio drivers. You can try other audio tools, e.g., https://github.com/wenxuanjun/jack-starter, or use another laptop.
